File: 520-detect-capital/detect-capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        allCap = all( i.isupper() for i in word)
        allLower = all(i.islower() for i in word)
        StoryCap = word.istitle()

        return allCap or allLower or StoryCap
        

File: 520-detect-capital/detect-capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) 
    {
        bool isAllCaps = true;
        for(char i : word)
        {
            if ( !isupper(i)) 
            {
                isAllCaps = false;
                break;

            }
        }

        bool isAllLower = true;
        for(char i : word)
        {
            if ( !islower(i) ) 
            {
                isAllLower = false;
                break;

            }
        }

        bool isTitleCase = false;
        if ( isupper(word[0] )) isTitleCase = true;
        for(int i=1 ; i < word.size() ;i++)
        {
            if ( !islower(word[i]))
            {
                isTitleCase = false;
                break;
            }
        }

        return isAllCaps || isTitleCase ||isAllLower ;
        
    }
};